hubmap_commons/hm_auth.py

```diff
@@ -485,23 +507,26 @@ def getHMGroups():
             if AuthCache.groupLastRefreshed is not None:
                 diff = now - AuthCache.groupLastRefreshed
             if diff is None or diff.days > 0 or diff.seconds > TOKEN_EXPIRATION:
-                groupIdByName = {}                    
-                #groupsById = {}                    
-                with open(AuthCache.groupJsonFilename) as jsFile:
-                    groups = json.load(jsFile)
-                    for group in groups:
-                        if 'name' in group and 'uuid' in group and 'generateuuid' in group and 'displayname' in group and not string_helper.isBlank(group['name']) and not string_helper.isBlank(group['uuid']) and not string_helper.isBlank(group['displayname']):
-                            group_obj = {'name' : group['name'].lower().strip(), 'uuid' : group['uuid'].lower().strip(),
-                                         'displayname' : group['displayname'], 'generateuuid': group['generateuuid']}
-                            if 'tmc_prefix' in group:
-                                group_obj['tmc_prefix'] = group['tmc_prefix']
-                            if 'data_provider' in group:
-                                group_obj['data_provider'] = group['data_provider']
-                            if 'shortname' in group:
-                                group_obj['shortname'] = group['shortname']
-                            groupIdByName[group['name'].lower().strip()] = group_obj
-                            AuthCache.groupsById[group['uuid']] = group_obj
-            return groupIdByName
+                if AuthCache.globusGroups is not None:
+                    return AuthCache.globusGroups
+                else:
+                    groupIdByName = {}
+                    #groupsById = {}
+                    with open(AuthCache.groupJsonFilename) as jsFile:
+                        groups = json.load(jsFile)
+                        for group in groups:
+                            if 'name' in group and 'uuid' in group and 'generateuuid' in group and 'displayname' in group and not string_helper.isBlank(group['name']) and not string_helper.isBlank(group['uuid']) and not string_helper.isBlank(group['displayname']):
```

Comment:
Related to above comment, better to use a function call instead of duplicating the logic.

Reply:
@yuanzhou  This should be resolved in latest commit